flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaWriterITCase.java

```diff
@@ -200,4 +270,96 @@ public UserCodeClassLoader getUserCodeClassLoader() {
             throw new UnsupportedOperationException("Not implemented.");
         }
     }
+
+    private static class DummySinkWriterContext implements SinkWriter.Context {
+        @Override
+        public long currentWatermark() {
+            return 0;
+        }
+
+        @Override
+        public Long timestamp() {
+            return null;
+        }
+    }
+
+    private static class NumBytesOutOperatorIOMetricGroup implements OperatorIOMetricGroup {
```

Comment:
Could you please check if you can use `UnregisteredMetricGroups.createUnregisteredOperatorMetricGroup().getIOMetricGroup()` instead? For another idea, see below.